diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/IndentationRule.kt

```diff
@@ -255,27 +262,66 @@ class IndentationRule(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     /**
-     * If it is triple-quoted string template we need to indent all its parts
+     * Indents each [entry][LITERAL_STRING_TEMPLATE_ENTRY] in a (potentially,
+     * multi-line) triple-quoted [string template][STRING_TEMPLATE].
+     *
+     * String templates usually have the following structure:
+     *
+     * * `STRING_TEMPLATE`
+     *    * `OPEN_QUOTE`
+     *    * `LITERAL_STRING_TEMPLATE_ENTRY`
+     *       * `REGULAR_STRING_PART`
+     *    * &#x2026;
+     *    * `LITERAL_STRING_TEMPLATE_ENTRY`
+     *       * `REGULAR_STRING_PART`
+     *    * `CLOSING_QUOTE`
+     *
+     * @param stringTemplate the string template.
+     * @see STRING_TEMPLATE
+     * @see LITERAL_STRING_TEMPLATE_ENTRY
      */
     @Suppress("LOCAL_VARIABLE_EARLY_DECLARATION")
     private fun fixStringLiteral(
         stringTemplate: ASTNode,
         expectedIndent: Int,
         actualIndent: Int
     ) {
-        val textIndent = " ".repeat(expectedIndent + INDENT_SIZE)
         val templateEntries = stringTemplate.getAllChildrenWithType(LITERAL_STRING_TEMPLATE_ENTRY)
-        templateEntries.forEach { node ->
-            if (!node.text.contains("\n")) {
-                fixFirstTemplateEntries(node, textIndent, actualIndent)
+        templateEntries.asSequence().filterIndexed { index, templateEntry ->
+            val text = templateEntry.text
+            val containsNewline = text.contains(NEWLINE)
+
+            if (containsNewline) {
+                /*
+                 * In real-life cases observed, whenever a `LITERAL_STRING_TEMPLATE_ENTRY`
+                 * _contains_ a newline character, it is _exactly_ a newline character.
+                 */
+                check(text.length == 1) {
+                    val escapedText = text.replace(NEWLINE.toString(), "\\n")
```

Comment:
maybe `\` + NEWLINE instead of '\n'?

Reply:
Tempting as it is, it merely won't compile =)